pkg/ignition/ignition.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package ignition
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/precond"
+)
+
+var ignitionLocalDir = "/var/run/libvirt/ignition-dir"
+
+const IgnitionFile = "data.ign"
+
+func GetIgnitionSource(vmi *v1.VirtualMachineInstance) string {
+	precond.MustNotBeNil(vmi)
+	return vmi.Annotations[v1.IgnitionAnnotation]
+}
+
+// func SetIgnitionLocalDirectory(dir string) error {
```

Comment:
I think we'll need this function for consistency with how our other user data generation works. 

take a look at cmd/virt-launcher.go's ```initializeDirs()``` function. We want to be able to dynamically set the ephemeral data dir.